File: JANUARY/08-01-2026/Count Subarray with k odds.py
class Solution:
    def countSubarrays(self, arr, k):
        from collections import defaultdict
        
        count_map = defaultdict(int)
        count_map[0] = 1
        prefix_sum = 0
        result = 0
        
        for num in arr:
            if num % 2 != 0:
                prefix_sum += 1
            
            result += count_map[prefix_sum - k]
            
            count_map[prefix_sum] += 1
        
        return result

File: 2026_GFG_POTD/JANUARY/08-01-2026/Count Subarray with k odds.cs
using System;
using System.Collections.Generic;

class Solution
{
    public int countSubarrays(int[] arr, int k)
    {
        Dictionary<int, int> countMap = new Dictionary<int, int>();
        countMap[0] = 1;
        int prefixSum = 0;
        int result = 0;
        foreach (int num in arr)
        {
            if (num % 2 != 0)
                prefixSum += 1;
            if (countMap.ContainsKey(prefixSum - k))
                result += countMap[prefixSum - k];
            if (countMap.ContainsKey(prefixSum))
                countMap[prefixSum]++;
            else
                countMap[prefixSum] = 1;
        }
        return result;
    }
}